test/options/identifierCase.ts

```diff
@@ -0,0 +1,71 @@
+import dedent from 'dedent-js';
+
+import { FormatFn } from '../../src/sqlFormatter.js';
+
+export default function supportsIdentifierCase(format: FormatFn) {
+  it('preserves identifier case by default', () => {
+    const result = format('select Abc from tBl1 left join Tbl2 where colA > 1 and colB = 3');
+    expect(result).toBe(dedent`
+      select
+        Abc
+      from
+        tBl1
+        left join Tbl2
+      where
+        colA > 1
+        and colB = 3
+    `);
+  });
+
+  it('converts identifiers to uppercase', () => {
+    const result = format('select Abc from tBl1 left join Tbl2 where colA > 1 and colB = 3', {
+      identifierCase: 'upper',
+    });
+    expect(result).toBe(dedent`
+      select
+        ABC
+      from
+        TBL1
+        left join TBL2
+      where
+        COLA > 1
+        and COLB = 3
+    `);
+  });
+
+  it('converts identifiers to lowercase', () => {
+    const result = format('select Abc from tBl1 left join Tbl2 where colA > 1 and colB = 3', {
+      identifierCase: 'lower',
+    });
+    expect(result).toBe(dedent`
+      select
+        abc
+      from
+        tbl1
+        left join tbl2
+      where
+        cola > 1
+        and colb = 3
+    `);
+  });
+
+  it('does not uppercase identifiers inside strings', () => {
```

Comment:
nit: I wouldn't really call it an identifier inside a string - it's just a string.

I personally wouldn't add a separate test for this, I'd just include some strings inside the general "converts identifiers to lowercase/uppercase" test cases. But that's really a personal preference. Doesn't matter much either way.

Reply:
String test is now included in identifier case tests and removed as separate test.

Reply:
String test is now included in identifier case tests and removed as separate test.